plugin/src/main/java/org/openrewrite/gradle/RewriteReflectiveFacade.java

```diff
@@ -193,6 +193,160 @@ public String print() {
                 throw new RuntimeException(e);
             }
         }
+
+        public SourceFile addJavaProvenance(JavaProvenance provenance) {
+            //Reflectively calling sourcefile.withMarkers(sourcefile.getMarkers().addIfAbsent(provenance)
+            try {
+
+                return new SourceFile(sourceFileWithMarkersMethod().invoke(this.real,
+                        markersAddIfAbsentMethod().invoke(sourceFileGetMarkersMethod().invoke(this.real), provenance.real)
+                ));
+            } catch (Exception e) {
+                throw new RuntimeException(e);
+            }
+        }
+    }
+    private Method sourceFileGetMarkers = null;
+    private Method sourceFileGetMarkersMethod() {
+        if (sourceFileGetMarkers == null) {
+            try {
+                Class<?> sourceFileClass = getClassLoader().loadClass("org.openrewrite.SourceFile");
+                sourceFileGetMarkers = sourceFileClass.getMethod("getMarkers");
+            } catch (Exception e) {
+                throw new RuntimeException(e);
+            }
+        }
+        return sourceFileGetMarkers;
+    }

```

Comment:
Why isn't this `RewriteReflectiveFacade.SourceFile.getMarkers()`? Same question re: `sourceFileWithMarkersMethod()` and `markersAddIfAbsentMethod()`.

This is inconsistent with how the rest of the facade is laid out and I don't see why that should be.

Reply:
I honestly was trying to avoid creating:

A facade for Markers, and a polymorphic facade for Marker and then having JavaProvenance extend that. 

When in fact we know that there is a very limited use case for adding the JavaProvenance